pinecone/client.go

```diff
@@ -759,6 +759,84 @@ func (c *Client) DeleteIndex(ctx context.Context, idxName string) error {
 	return nil
 }
 
+// ConfigureIndex is used to [scale a pods-based index] up or down by changing the size of the pods or the number of
+// replicas.
+//
+// Parameters:
+//   - name: The name of the index to configure.
+//   - pods: (Optional) The pod size to scale the index to (e.g. for a "p1" pod type,
+//     you could pass "p1.x2" to scale your index to the "x2" size,
+//     or you could pass "p1.x4" to scale your index to the "x4" size, and
+//     so forth.
+//   - replicas: (Optional) The number of replicas to scale the index to.
+//     This is capped by the maximum number of replicas allowed in your Pinecone project. To configure this number,
+//     go to [app.pinecone.io], select your project, and configure the maximum number of pods.
+//
+// Note: You can only scale an index up, not down. If you want to scale an index down,
+// you must create a new index with the desired configuration.
+//
+// Returns a pointer to a configured Index object or an error.
+//
+// Example for a pods-based index originally configured with 1 "p1" pod of size "x2" and 1 replica:
+//
+//	// To scale the size of your pods from "x2" to "x4":
+//	 _, err := pc.ConfigureIndex(ctx, "my-index", "p1.x4", nil)
+//	 if err != nil {
+//	     fmt.Printf("Failed to configure index: %v\n", err)
+//	 }
+//
+//	// To scale the number of replicas:
+//	 _, err := pc.ConfigureIndex(ctx, "my-index", nil, 4)
+//	 if err != nil {
+//	     fmt.Printf("Failed to configure index: %v\n", err)
+//	 }
+//
+//	// To scale both the size of your pods and the number of replicas:
+//	 _, err := pc.ConfigureIndex(ctx, "my-index", "p1.x4", 4)
+//	 if err != nil {
+//	     fmt.Printf("Failed to configure index: %v\n", err)
+//	 }
+//
+// [scale a pods-based index]: https://docs.pinecone.io/guides/indexes/configure-pod-based-indexes
+// [app.pinecone.io]: https://app.pinecone.io
+func (c *Client) ConfigureIndex(ctx context.Context, name string, podType *string,
+	replicas *int32) (*Index, error) {
+
+	if podType == nil && replicas == nil {
+		return nil, fmt.Errorf("must specify either podType or replicas")
+	}
+
+	request := control.ConfigureIndexRequest{
+		Spec: struct {
+			Pod struct {
+				PodType  *control.PodSpecPodType  `json:"pod_type,omitempty"`
+				Replicas *control.PodSpecReplicas `json:"replicas,omitempty"`
+			} `json:"pod"`
+		}{
+			Pod: struct {
+				PodType  *control.PodSpecPodType  `json:"pod_type,omitempty"`
+				Replicas *control.PodSpecReplicas `json:"replicas,omitempty"`
+			}{
+				PodType:  podType,
+				Replicas: replicas,
+			},
+		},
+	}
+
+	res, err := c.restClient.ConfigureIndex(ctx, name, request)
+	if err != nil {
+		return nil, err
+	}
+
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
```

Comment:
Changed this from our usual`http.StatusCreated` to `http.StatusOK`, since configuring an index doesn't give status created or whatnot. Just fyi!